url-inspection/app/controllers/inspectionWorker.js

```diff
@@ -52,8 +53,23 @@ db.mongoose
   });
 
 startLinkInspection = async (url, inspectedLink) => {
-  const inspectLinkObj = await new InspectLinks(inspectedLink).save(); // add new entry to DB with status "processing"
-  inspectedLink._id = inspectLinkObj["_id"];
+
+  var inspectLinkObj = null;
+  var urlPreviouslyInspected = false;
+
+  inspectLinkObj = await InspectLinks.findOne({ "original_url": url });
+
+  if (inspectLinkObj != null) {
+    urlPreviouslyInspected = true;
+    inspectedLink._id = inspectLinkObj["_id"];
+    inspectedLink.to_flag = inspectLinkObj["to_flag"];
+    inspectLinkObj = await InspectLinks.findOneAndUpdate({ "original_url": url }, inspectedLink);
```

Comment:
this just updates an existing record instead of creating a new one, i dont think there will be any changes on the frontend cuz its essentially still doing the inspect

i think previously if another record of the same link already existed, the record pulled from db on the frontend wasn't the latest one right?